vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/JcrPackageRegistry.java

```diff
@@ -655,8 +655,8 @@ public void remove(@NotNull PackageId id) throws IOException, NoSuchPackageExcep
         if (pkg == null) {
             throw new NoSuchPackageException().setId(id);
         }
-        JcrPackage pack = pkg.getJcrPackage();
         try {
+            JcrPackage pack = pkg.getJcrPackage();
```

Comment:
you can use [try with resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) as `JcrPackage` implements `AutoClosable`.

Reply:
Ah, missed that one. Though, it seems it is not necessary in this case. Closing `JcrRegisteredPackage` already closes the `JcrPackage` it holds.

Reply:
Done. Also put the `snap` `JcrPackage` into a try with resources.